### ISSUE TITLE
A new linter checking duplicate arguments

### ISSUE THREAD

**Issue by renkun-ken** [jimhester/lintr#850]
It is almost always a mistake to call a function with duplicate arguments, i.e. arguments of the same name, like the following:

```r
foo(bar1 = 1, bar1 = 2, bar3 = 3)
```

Therefore, in most cases, it might make sense to check all function calls and ensure there are no duplicate arguments.

One use case of duplicate argument is to produce some table that actually needs duplicate headers for demonstration in documents, e.g.

```r
data <- data.frame(
  mean = 1:5,
  sd = 0.5,
  mean = 2:6,
  sd = 1,
  check.names = FALSE
)
```

**Comment by MichaelChirico** [comment]
do we want to maintain an allowlist of functions where duplicates are allowed? data.frame(), data.table(), c(), list(), tibble() & compatriots, ...

**Comment by renkun-ken** [comment]
I think just like `missing_argument_linter(except = c(...))` would be enough. But I suspect even these functions should not be allowed by default too because in most scenarios using duplicate arguments with these functions are undesired too.
